examples/tests/linuxbuild.py

```diff
@@ -17,24 +17,31 @@ class LinuxBuildTest(Test):
     """
 
     def setUp(self):
+        kernel_src_url = self.params.get('linux_src_url', default=None)
         kernel_version = self.params.get('linux_version', default='3.19.8')
         linux_config = self.params.get('linux_config', default=None)
+        self.do_kernel_install = self.params.get('do_kernel_install', default=None)
+        if kernel_src_url is None:
```

Comment:
The `avocado.utils.kernel` library is supposed to be able to use the default kernel.org repo to download sources based on the given version number.  This effectively disables that capability.  The kernel source URL should be given to `kernel.KernelBuild` if present in the parameters, or the default should be used.

Reply:
The issue we ran into is that all kernel versions are not available from a single url. ei:
For v3.x kernels, the following url must be used:
https://www.kernel.org/pub/linux/kernel/v3.x/
For v4.x kernels, the following url must be used:
https://www.kernel.org/pub/linux/kernel/v4.x/

Due to having to use different url for different major kernel versions, we thought providing the url as parameter could be one way around.

Reply:
See my previous comment about the `url` parameter to `download()`.  Also, the default `kernel.KernelBuild` will get the right URL based on the kernel version.  This was fixed recently:

http://avocado-framework.readthedocs.io/en/latest/release_notes/59_0.html#bug-fixes

Reply:
Thank you,
I am just noticing the fix.

Would it be ok to keep the url functionality of this pull-request for cases where we need to pull the kernel to build from a different location ?

Reply:
You mean adding a `url` parameter to `kernel.KernelBuild.donwload()` API? That's a nice improvement.  Just account for the two situations (passing a URL and using the default one).